### PR TITLE
Implementation of inactive regions along partial wafer cuts 

### PR DESCRIPTION
**Pull request by Pruthvi-ch**
PR description:

Implement script to reject hits from the inactive zones along partial wafer cuts of HGCAL. (only reject for partial wafers, Full wafers left unaffected)

PR validation:

Use runTheMatrix test workflows 'runTheMatrix.py -l limited -i all --ibeos'
Validation results presented in: https://indico.cern.ch/event/1297259/

if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by Pruthvi-ch** [comment]
There is a issue in edm log, New commit has been made to correct the issue

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: SimG4CMS/Calo/src/HGCGuardRingPartial.cc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      double delX = 0.5 * waferSize_;
+      double delY = 2 * delX / sqrt3_;
+      double dx = (zside > 0) ? -point.x() : point.x();
+      double dy = point.y();
+      if (type > 0) {
+        check = (std::abs(dy - (dx * tan_1[placement])) < std::abs(offset_ / cos_1[placement]) || check);
+        check = (std::abs(dy - (dx * tan_1[placement]) + ((HGCalTypes::c10 * delY * 0.5) / cos_1[placement])) <
+                     std::abs(offset_ / cos_1[placement]) ||
+                 check);
+        check = (std::abs(dy * cot_1[placement] - (dx)) < std::abs(offset_ / cos_1[placement]) || check);
```
Comment (perrotta) [inline SimG4CMS/Calo/src/HGCGuardRingPartial.cc:42]:
```suggestion
        double thresh = std::abs(offset_ / cos_1[placement])
        check |= std::abs(dy - (dx * tan_1[placement])) < thresh;
        check |= std::abs(dy - (dx * tan_1[placement]) + (HGCalTypes::c10 * delY * 0.5) / cos_1[placement]) <
                    thresh;
        check |= std::abs(dy * cot_1[placement] - dx) < thresh;
```
can avoid a few useless repeated calculation

File: SimG4CMS/Calo/src/HGCGuardRingPartial.cc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      } else {
+        check = (std::abs((dy * cot_1[placement]) - dx + ((HGCalTypes::c22 * delX) / cos_1[placement])) <
```
Comment (perrotta) [inline SimG4CMS/Calo/src/HGCGuardRingPartial.cc:44]: same here

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**